nixos/modules/services/web-apps/tachidesk-server.nix

```diff
@@ -0,0 +1,208 @@
+{ config, pkgs, lib, ... }:
+
+let
+  cfg = config.services.tachidesk-server;
+  inherit (lib) mkOption mdDoc mkEnableOption mkIf types;
+in {
+  options = {
+    services.tachidesk-server = {
+      enable = mkEnableOption (mdDoc "A free and open source manga reader server that runs extensions built for Tachiyomi.");
+
+      package = lib.mkPackageOptionMD pkgs "tachidesk-server" { };
+
+      dataDir = mkOption {
+        type = types.path;
+        default = "/var/lib/tachidesk-server";
+        description = mdDoc ''
+          The path to the data directory in which Tachidesk will download scans.
+        '';
+      };
+
+      user = mkOption {
+        type = types.str;
+        default = "tachidesk";
+        description = mdDoc ''
+          User account under which Tachidesk runs.
+        '';
+      };
+
+      group = mkOption {
+        type = types.str;
+        default = "tachidesk";
+        description = mdDoc ''
+          Group under which Tachidesk runs.
+        '';
+      };
+
+      openFirewall = mkOption {
+        type = types.bool;
+        default = false;
+        description = mdDoc ''
+          Whether to open the firewall for the port in {option}`services.tachidesk-server.config.port`.
+        '';
+      };
+
+      config = {
+        ip = mkOption {
+          type = types.str;
+          default = "0.0.0.0";
+          description = mdDoc ''
+            The ip that Tachidesk will bind to.
+          '';
+        };
+
+        port = mkOption {
+          type = types.port;
+          default = 8080;
+          description = mdDoc ''
+            The port that Tachidesk will listen to.
+          '';
+        };
+
+        basicAuth = {
+          enable = mkOption {
+            type = types.bool;
+            default = false;
+            description = mdDoc ''
+              Controls whether if Tachidesk-Server should be protected with Basic access authentication.
+              Enabling this option is useful when hosting on a public network/the Internet.
+            '';
+          };
+
+          username = mkOption {
+            type = types.str;
+            description = mdDoc ''
+              The username value that you have to provide when authenticating.
+            '';
+          };
+
+          passwordFile = mkOption {
+            type = types.path;
```

Comment:
question: should this have a default option (if there's no appropriate default, can the type be a union with null)?

ditto for `username`?

Reply:
I don't think putting a default is a good option here so I change the type to `nullOr path`.

Reply:
I suppose you could make the default `null` and throw an error if it is `null`?

Reply:
Oh yeah it makes sense, I change this!